src/main/scala/temple/ast/StructBlock.scala

```diff
@@ -2,8 +2,7 @@ package temple.ast
 
 import temple.ast.Metadata.StructMetadata
 
-/** A nested struct within a service, representing a second table/domain in the same datastore */
-case class StructBlock(
-  attributes: Map[String, Attribute],
-  metadata: Seq[StructMetadata] = Nil,
-) extends TempleBlock[StructMetadata]
+trait StructBlock[M >: StructMetadata <: Metadata] extends TempleBlock[M] {
```

Comment:
Hmm I'm not sure about the naming `Struct` and `NestedStruct`. I feel like `Struct` needs to stay as is (to match Templefile syntax) and this becomes something `AbstractStruct` to make it clear it's a trait? Not sure about names though